modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -329,51 +321,100 @@ impl ServerModule for Wallet {
                     signature: val,
                 })
             })
-            .chain(stream::once(async { round_ci }))
             .collect::<Vec<WalletConsensusItem>>()
             .await;
 
-        // We force new epochs only if height changed, or we have peg-outs (more than
-        // just round_ci item)
-        if last_consensus_height < proposed_height || 1 < items.len() {
-            ConsensusProposal::Trigger(items)
-        } else {
-            ConsensusProposal::Contribute(items)
+        let block_height_proposal = self
+            .block_height()
+            .await
+            .saturating_sub(self.cfg.consensus.finality_delay);
+
+        if block_height_proposal != self.consensus_block_height(dbtx).await {
+            items.push(WalletConsensusItem::BlockHeight(block_height_proposal));
         }
+
+        let fee_rate_proposal = self.fee_rate().await;
+
+        if fee_rate_proposal != self.consensus_fee_rate(dbtx).await {
+            items.push(WalletConsensusItem::Feerate(fee_rate_proposal));
+        }
+
+        ConsensusProposal::new_auto_trigger(items)
     }
 
     async fn begin_consensus_epoch<'a, 'b>(
         &'a self,
         dbtx: &mut ModuleDatabaseTransaction<'b>,
         consensus_items: Vec<(PeerId, WalletConsensusItem)>,
-        consensus_peers: &BTreeSet<PeerId>,
+        _consensus_peers: &BTreeSet<PeerId>,
     ) -> Vec<PeerId> {
         trace!(?consensus_items, "Received consensus proposals");
 
-        // Separate round consensus items from signatures for peg-out tx. While
-        // signatures can be processed separately, all round consensus items
-        // need to be available at once.
-        let UnzipWalletConsensusItem {
-            peg_out_signature: peg_out_signatures,
-            round_consensus: round_items,
-        } = consensus_items.into_iter().unzip_wallet_consensus_item();
```

Comment:
I wonder where this is still used now, would be nice if we could remove it (iirc the impl is generated by a proc macro)